### PR TITLE
mention inline variable declaration

### PR DESCRIPTION
**Pull request by ePaul**
This hopefully fixes an issue (the non-mention of the declaration) brought up by @Zambonifofex in Gitter channel ceylon/user.

I guess this example wasn't the best one, feel free to propose a better one.

### CODE REVIEW

File: documentation/1.2/reference/statement/switch.md [hunk 1]
```diff
@@ -144,6 +144,23 @@ clause of the control structure:
         }
     }
 
+### Variable declaration
+
+Instead of the `switch` expression also an inline variable declaration (containing an initializing expression) is allowed.
+This declares a new variable `name` which is then usable inside the case and else blocks (with the correct type in each of them).
+
+    switch(line = Process.readLine())
```
Comment (lucaswerkmeister) [inline documentation/1.2/reference/statement/switch.md]: s/Process/**p**rocess/

Comment (ePaul) [inline documentation/1.2/reference/statement/switch.md]: Argh. I really should have tested this first.

**Comment by gavinking** [comment]
@ePaul this is better language:

> As an alternative to the `switch` expression, an inline variable declaration (containing an initializing expression) is allowed.

**Comment by ePaul** [comment]
@gavinking Thanks, fixed.

**Comment by gavinking** [comment]
Now: 

> This example declares a new variable `name` which is then usable inside the `case` and `else` blocks (with the correct type in each of them).

1. "This example" makes it clear that "this" refers to the code example that comes next, not the previous sentence.
2. backticks around the keywords.

:-)

I'm an absolute style nazi as far as docs go. :)

**Comment by ePaul** [comment]
> 1. "This example" makes it clear that "this" refers to the code example that comes next, not the previous sentence.

Actually I meant to refer to the previous sentence, not to the example. (I also meant to declare a "variable name", not a "variable named `name`" ... I added the backticks after @bjansen's comment, missing the `case` and `else` keywords.)

When referring to the example, I guess instead of `name` we should use `line` here.

**Comment by gavinking** [comment]
> Actually I meant to refer to the previous sentence, not to the example. (I also meant to declare a "variable name", not a "variable named name" 

Ah ok ... so you see, it *was* confusing: I was confused by it ;-)

**Comment by ePaul** [comment]
So my alternative proposal for the second sentence would be this:

> This declares a new value which is then usable inside the `case` and `else` blocks (with the correct type in each of them).

(Also changed variable → value, because it is not actually changeable.)

**Comment by gavinking** [comment]
I think "a new variable" was fine (that's the language the spec uses for these things). You just didn't need the word "name" in there. :)

**Comment by gavinking** [comment]
Yay! :-)

**Pull request merged**